operators/src/source/gdal_source/mod.rs

```diff
@@ -733,7 +734,27 @@ where
             loading_info.info
         };
 
-        let source_stream = stream::iter(loading_iter);
+        let loading_inter_checked = loading_iter.with_position().map(move |(pos, ele)| {
```

Comment:
so you go through all loading infos just to check the first and last? Wouldn't it suffice to lazily check the loading infos as the are consumed or is that to complicated?

Reply:
changed that and the providers using lists and/or generators now iterate while generating. The only exception is the metadata regular which must iterate the time steps in advance. This is needed since we need the time bounds when initializing the adapter